packages/snaps-controllers/src/snaps/registry/json.ts

```diff
@@ -153,6 +156,21 @@ export class JsonSnapsRegistry extends BaseController<
     );
   }
 
+  async #triggerUpdate() {
+    // If an update is ongoing, wait for that.
+    if (this.#currentUpdate) {
+      await this.#currentUpdate;
+      return;
+    }
+    // If no update exists, create promise and store globally.
+    if (this.#currentUpdate === null) {
+      this.#currentUpdate = this.#update();
+    }
+    await this.#currentUpdate;
+    this.#currentUpdate = null;
+  }
+
+  // Should not be called directly, instead triggerUpdate should be used.
```

Comment:
It would be better if this function have a docstring and this comment added as a note. That way it's more likely that it would be respected in the future, since most editors will show the notes/docstring in a popup. Having just a comment above can be invisible in some cases while maintaining.

Reply:
Yeah, good idea, done!